chess/src/test/scala/io/github/chess/engine/model/rules/chess/king/KingRuleSpec.scala

```diff
@@ -6,25 +6,117 @@
  */
 package io.github.chess.engine.model.rules.chess.king
 
-import io.github.chess.AbstractSpec
-import io.github.chess.engine.model.board.{ChessBoard, Position}
-import io.github.chess.engine.model.game.{ChessGameStatus, Team}
-import io.github.chess.engine.model.pieces.{King, Pawn}
-
-/** Test suite for [[King]]. */
-class KingRuleSpec extends AbstractSpec:
-
-  private val initialPosition: Position = (1, 2)
-  private val exactPositions: Set[Position] =
-    Set((0, 2), (0, 3), (1, 1), (1, 3), (2, 1), (2, 2), (2, 3))
-  private val rule = KingMovementRule()
-  private val chessBoard: ChessBoard =
-    ChessBoard.empty
-      .setPiece((0, 1), Pawn(Team.WHITE))
-      .setPiece(initialPosition, King(Team.WHITE))
-  private val chessGameStatus = ChessGameStatus(chessBoard)
-  private val moves = rule.findMoves(initialPosition, chessGameStatus).map(_.to)
-
-  "The KingMovementRule" should "let move the king in all the directions stepped by one" in {
-    moves shouldEqual exactPositions
+import io.github.chess.engine.model.board.ChessBoardBuilder.DSL.*
+import io.github.chess.engine.model.board.{ChessBoard, File, Position, Rank}
+import io.github.chess.engine.model.game.ChessGameStatus
+import io.github.chess.engine.model.rules.AbstractChessRuleSpec
+
+/** Test suite for [[KingRule]]. */
+class KingRuleSpec extends AbstractChessRuleSpec:
+
+  private val kingPosition = Position(File.E, Rank._5)
+
+  private val chessBoardOnlyKing = ChessBoard {
+    * | * | * | * | * | * | * | *
+    * | * | * | * | * | * | * | *
+    * | * | * | * | * | * | * | *
+    * | * | * | * | K | * | * | *
+    * | * | * | * | * | * | * | *
+    * | * | * | * | * | * | * | *
+    * | * | * | * | * | * | * | *
+    * | * | * | * | * | * | * | *
+  }
+  private val chessBoardMovesOnlyKing = ChessBoard {
+    * | * | * | * | * | * | * | *
+    * | * | * | * | * | * | * | *
+    * | * | * | X | X | X | * | *
+    * | * | * | X | * | X | * | *
+    * | * | * | X | X | X | * | *
+    * | * | * | * | * | * | * | *
+    * | * | * | * | * | * | * | *
+    * | * | * | * | * | * | * | *
+  }
+
+  "The king rule" should "allow the knight to move in all the directions stepped by one" in {
```

Comment:
Typo in KingRuleSpec at line 40: "knight" should be "king"

Reply:
Done. Ty